### PR TITLE
Fix bug in mmult for Trilinos matrix

### PR DESCRIPTION
**Pull request by Rombur**
@dalg24 found a bug in `TrilinosWrappers::SparseMatrix::mmult` in parallel for some sparse matrices. The current code was pretty complicated so I just replaced it by `EpetraExt::MatrixMatrix::Multiply`. This means that we now require `EpetraExt` when compiling Trilinos but that allows us to remove more than one hundred lines of code and it fixes the bug. Of course this means that all the testers will crash if  they don't have `EpetraExt`.

The test is a little bit strange because I used one matrix that I know didn't work. So I just set the value of every entry by hand.

### CODE REVIEW

File: doc/news/changes/minor/20180822BrunoTurcksin [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Fixed: mmult using TrilinosWrappers::SparseMatrix sometimes gave the wrong
+result in parallel
```
Comment (bangerth) [inline doc/news/changes/minor/20180822BrunoTurcksin]: Write this as "TrilinosWrappers::SparseMatrix::mmult() sometimes gave the wrong result in parallel." to allow doxygen to link to the correct function.

**Comment by Rombur** [comment]
@masterleinad I did what you asked but I can't get the tests enabled even when I have `EpetraExt`. Any idea?

**Comment by Rombur** [comment]
OK this should work now.

**Comment by masterleinad** [comment]
You could also just have renamed the file to `sparse_matrix_mmult_02.with_trilinos_with_epetraext\=on.mpirun\=2.output`.

**Comment by Rombur** [comment]
Oh I tried `with_epetraext` alone and it didn't work that's why I create the new directory

**Comment by Rombur** [comment]
Can we merge this? I am want to rebuild my Docker images and I need this fix.

**Comment by Rombur** [comment]
Thanks @masterleinad

**Pull request merged**